samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -667,16 +666,29 @@ static int pa_sync_req_cb(struct bt_conn *conn,
 	}
 
 	if (IS_ENABLED(CONFIG_BT_PER_ADV_SYNC_TRANSFER_RECEIVER) && past_avail) {
+		int err;
+
 		err = pa_sync_past(conn, pa_interval);
+		if (err != 0) {
+			printk("Failed to subscribe to PAST: %d\n", err);
+
+			return err;
+		}
+
 		k_sem_give(&sem_past_request);
-	} else {
-		/* start scan */
-		err = 0;
+
+		err = bt_bap_scan_delegator_set_pa_state(recv_state->src_id,
```

Comment:
should we set the PA state to BT_BAP_PA_STATE_INFO_REQ to request PAST prior to calling pa_sync_past() in the line 671 ?

Reply:
> should we set the PA state to BT_BAP_PA_STATE_INFO_REQ to request PAST prior to calling pa_sync_past() in the line 671 ?

Hmm, good question, but I would argue no. If we (for whatever reason) fail to subscribe to PAST (which is what `pa_sync_past` does), then we don't need to update the state. I think we makes sense to update the state, when we reach a point that we are waiting for the remote device

Reply:
how about if we are not allowed to set the state to be BT_BAP_PA_STATE_INFO_REQ for some reason (e.g Could not find recv_state by src_id ....), do we still need to subscribe to PAST by pa_sync_past() ?

Reply:
Hmm, that is a fair point. 

So basically we have the option to do

1. Subscribe to past
2. Set state (if this fails, no state change was done, and we can return error)

or

1. Set state (if this fails, no state change was done, and we can return error)
2. Subscribe to past
3. If 2. fails, we need to restore state (and what if this fails?)

I would be OK with changing it to the latter, but it is a more complex solution, where as subscription to PAST is not really hurtful (and cannot be undone), even if we don't receive PAST from the peer. 

Let me know if you want me to change it, but my immediate opinion would be to keep as is to keep it simple

Reply:
I see, it is ok for me to choose option 1 :-)